test/support/utils.h

```diff
@@ -123,6 +130,52 @@ is_equal_val(const T1& val1, const T2& val2)
     }
 }
 
+template <typename T, typename TOutputStream, typename = void>
+struct IsSupportStreamOutput : std::false_type
+{
+};
+
+template <typename T, typename TOutputStream>
+struct IsSupportStreamOutput<T, TOutputStream,
+                             std::void_t<decltype(std::declval<TOutputStream>() << std::declval<T>())>> : std::true_type
+{
+};
+
+struct TagExpected{};
+struct TagActual{};
+
+inline
+std::string log_value_title(TagExpected)
+{
+    return " expected ";
+}
+
+inline
+std::string log_value_title(TagActual)
+{
+    return " got ";
+}
+
+template <typename TStream, typename Tag, typename TValue>
+ void log_value(TStream& os, Tag, const TValue& value, bool bCommaNeeded)
+{
+    if (bCommaNeeded)
+        os << ",";
+    os << log_value_title(Tag{});
+
+    if constexpr (IsSupportStreamOutput<TValue, decltype(os)>::value)
+    {
+        if constexpr (std::is_pointer_v<TValue>)
+            os << "0x";
```

Comment:
When I tried this out with a pointer, I got something like `... 0x0x15061eb0, got 0x0x15061eb4`. I do not know if the output for pointers is standardized or implementation defined, but it seems like libstdc++ at least already has the hexadecimal prefix.

Reply:
Fixed: "0x" has been removed